int32 nSkeletons
skeleton[] skeletons